docker_cli/_cli.py

```diff
@@ -0,0 +1,327 @@
+import argparse
+import sys
+import textwrap
+from typing import Sequence
+
+from ._utils import universal_tag_prefix
+from .commands import dropin, make_lockfile, remove
+from .setup_commands import (setup_all, setup_cuda_dev, setup_cuda_runtime,
+                             setup_env_add, setup_env_dev, setup_env_runtime,
+                             setup_init)
+
+
+def setup_parser() -> argparse.ArgumentParser:
+    """
+    Create a top-level argument parser.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        The parser.
+    """
+    prefix = universal_tag_prefix()
+
+    def help_formatter(prog): return argparse.HelpFormatter(
+        prog,
+        max_help_position=60
+    )
+
+    parser = argparse.ArgumentParser(
+        prog=__package__,
+        formatter_class=help_formatter
+    )
+
+    # Additional parsers for shared commands
+    setup_parse = argparse.ArgumentParser(add_help=False)
+    setup_parse.add_argument(
+        "--base", "-b", type=str, required=True,
+        help='The name of the base docker image.'
```

Comment:
In public-facing docstrings, when we refer to other projects like Docker, we should try to use proper capitalization of the project name:

```suggestion
        help='The name of the base Docker image.'
```

Other examples would be like "numpy" -> "NumPy", "isce3" -> "ISCE3", etc